src/stores/A_store.js

```diff
@@ -71,6 +71,53 @@ class A_store extends ReduceStore {
                 return {...state, stage, path, player: {x: nextLocation.x, y: nextLocation.y}};
             }
 
+            case ActionTypes.GENERATE_FOG: {
+                const {player, end} = state;
+                const {x, y} = player;
+                const revealedGrid = mapGenerator(player, end, new Grid(SIZE, SIZE));
+                const pathGrid = new Grid(SIZE, SIZE);
+
+                let visibleCells = revealedGrid.getVisible(x, y);
+                revealedGrid.obfuscateGrid(x, y, pathGrid, visibleCells);
+                let seenCells;
+                visibleCells = pathGrid.getVisible(x, y);
+                seenCells = pathGrid.getVisible(x, y);
+
+                const start = pathGrid.getCell(x, y);
+                const goal = pathGrid.getCell(end.x, end.y);
+                const path = Astar(start, goal, Cell.heuristic);
+                path.shift();
+
+                return {...state, pathGrid, revealedGrid, path, stage: 'STEP_FOG', visibleCells, seenCells};
+            }
+
+            case ActionTypes.STEP_FOG: {
+                let path = [...state.path];
+                const {revealedGrid, end, seenCells} = state;
+                const nextLocation = path.shift();
+                let stage, pathGrid, visibleCells;
+
+                if (path.length > 0) {
+                    stage = 'STEP_FOG';
+                    visibleCells = revealedGrid.getVisible(nextLocation.x, nextLocation.y);
+                    pathGrid = revealedGrid.obfuscateGrid(nextLocation.x, nextLocation.y, state.pathGrid, visibleCells);
+                    visibleCells = pathGrid.getVisible(nextLocation.x, nextLocation.y);
+                    for (const cell of visibleCells) {
+                        seenCells.add(cell);
+                    }
+
+                    const start = pathGrid.getCell(nextLocation.x, nextLocation.y);
+                    const goal = pathGrid.getCell(end.x, end.y);
+                    path = Astar(start, goal, Cell.heuristic);
+                    path.shift();
+                } else {
+                    stage = 'RESET';
+                    pathGrid = revealedGrid;
+                }
+
+                return {...state, stage, path, player: {x: nextLocation.x, y: nextLocation.y}, pathGrid, visibleCells, seenCells}
```

Comment:
i continue to think any object with more than a couple of properties probably ought to be on multiple lines

if you feel strongly about this then go ahead but you do this a ton and imo it makes it very hard to check at a glance what properties are involved

Reply:
🐇